### DISCUSSION TITLE
HA e Assistente vocale - qualche dubbio e una vostra opinione....

### DISCUSSION THREAD

**Discussion by Jocman** [indomus/forum#1821] [HUB: Home Assistant]
Ciao a tutti
Stavo studiacchiando qua e là in rete in tema HA, e sono un paio di giorni che mi sto soffermando sulle sue potenzialità…..comunicative.
Mi riferisco al fatto di poterlo far parlare e potergli parlare.
Tuttavia, mi sorgono varie perplessità a riguardo, soprattutto riguardo il metodo preferibile.
A questo premetto che sono “allergico” a tutto ciò che è cloud: preferirei cantarmela e suonarmela nell’intimità di casa mia, senza in qualche modo dover passare per terzi.
Premesso questo, le mie perplessità:
-	RhassPy: mi ha suscitato molto interesse questo sistema, mi sembra essere molto potente e versatile, ovviamente va….lavorato. Perplessità a riguardo: 
     - installarlo su un Raspi dedicato (tutta la potenza del raspi va a lui – anche se i raspi ultimamente sono aumentati un po’ di prezzo….), o come add-on di HA (ed in questo caso, l’HUB ne risentirebbe in termini di prestazioni – rallentamenti etc)?
     -	Dispositivi di interfacciamento (ovvero speaker / microfono): mi è parso di capire che le possibilità sono solo di elementi collegati direttamente al Raspi (vuoi via cavo, vuoi come board integrata); ma in questo caso, se io volessi mettere uno speaker/microfono in ogni ambiente???
     -	Il vantaggio, credo, è che il tutto non uscirebbe dalle mura di casa, quindi la mia “allergia al cloud” sarebbe salva
-	Alexa: credo sia la soluzione più diffusa (o sbaglio?). Tuttavia mi lascia perplesso il fatto di dover sottoscrivere servizi (per quanto gratuiti): 
     -	non significa comunque passare sempre per tramite terzi (AzD, AxDC, AWS) ogni volta che si “parla/ascolta”? (la mia “allergia”)
     -	riguardo a AWS Lambda, si sfrutterebbe la gratuità fino a 1 milione di richieste. Cosa si intende per “richiesta”?
     -	Il discorso speaker/microfono sparsi per casa in questo caso penso sarebbe risolto, però…..
Altro non mi pare di aver trovato (a parte google, ma tant’è….).
Mi farebbe piacere una opinione riguardo il tema, in modo anche da poter anche prendere una decisione nel momento in cui volessi realizzare la cosa.

**Comment by NdR91** [comment]
Il discorso "allergia al cloud" la tralascio. Il mondo non è Bianco o Nero, ci sono servizi validi e meno validi. Ci sono servizi dove si può giudicare un livello di... "sicurezza" più elevato e meno elevato. Nel 2022 volersi escludere totalmente dal cloud è quanto meno complesso. Se, aggiungo, lo si fa per una questione di "sicurezza", allora ritengo che l'unico modo per essere davvero sicuri è staccare il vavo LAN/Fibra dalla rete di casa.

Comunque è un tuo pensiero e non lo contesto, ho solo espresso la mia opinione.

Questione "Assistente vocale". Ci sono due considerazioni in merito ad Alexa:
1- L'assistente inteso come gestione della propria domotica personale (Alexa, accendi le luci...);
2- Avvisi di vario genere da inviare tramite Alexa (La lavatrice ha finito il ciclo di lavaggio!...);

Per il primo, ti confermo che è necessario passare da Alexa. Non ci sono altri modi. L'unica opzione disponibile consiste nel decidere se arrivarci tramite Nabu Casa (servizio a pagamento offerto dai dev di Home Assistant) o tramite AWS, creandoti tu la skill direttamente. Questo non cambia il risultato finale che interessa a te: la richiesta partirà dalla tua rete, per poi uscire verso i sistemi Amazon e poi rientrare nuovamente.

Per il secondo punto: il sistema di avvisi, notifiche e simili, è gestibile tutto in locale. Anche se, comunque, è necessario collegare l'integrazione (si chiama "Alexa Media Player") al proprio account Amazon. Questo perchè, comunque, la sincronizzazione dei Media Player disponibili (i vari Echo) avviene tramite account.

Non avendo mai approfondito RhassPy, non posso darti molte info a riguardo. Io stesso, nonostante fossi curioso, mi sono fermato in quanto è palese che le necessità in termini di hw richiedano molto più lavoro (microfoni, speaker...). 
Lavoro che io, personalmente, ho ritenuto ingiustificato a fronte dei 19€ ad Echo Dot.

Spero almeno di averti dato una risposta esaustiva lato Amazon Alexa.

**Reply by Jocman** [reply]
Grazie mille per la tua opinione (e per non aver giudicato la mia "allergia" - per inciso, sono consapevole che il mondo è in realtà "grigio", ma fin quando posso evitare cerco di farlo).
Circa Alexa, beh in effetti diciamo che hai confermato ciò che pensavo.
Francamente non so se sceglierò (quando sarà) l'opzione Alexa o RhassPy; come dici, a livello "portafoglio" Echo Dot è una soluzione bella e pronta ed economica. Tuttavia, avendo ancora tempo per decidere, cercherò di delucidarmi quanto più possibile la questione hardware del RhassPy (come detto, per quello che ho letto in giro, lo vedo molto flessibile e potente, a decidere se questa potenza e flessibilità valgono lo sforzo oggettivamente richiesto...)

**Reply by snakuzzo** [reply]
Comunque non esiste solo Alexa...
Dai un'occhiata anche a [QUESTO ARTICOLO](https://indomus.it/guide/local-fulfillment-integrare-localmente-google-assistant-a-home-assistant-addendum/)

**Reply by Jocman** [reply]
Aggiungo una piccola nota in autorisposta (sto continuando a studiare la documentazione): RhassPy può essere comandato da elementi satellite (l'equivalente di Alexa, insomma). Peccato che questi elementi satellite siano anche essi costituiti da un Raspi con la sezione audio (cassa+microfono). Insomma, si tratta di investire una media di 100€ o più a satellite (più tutto lo sbattimento implementativo e sempre che i prezzi del raspi non salgano ancora). Sinceramente, i 19€ full optional di Echo Dot sono un ottimo incentivo....

**Reply by snakuzzo** [reply]
io ti consiglio di leggere quello che ti ho linkato poco fa e pensare anche alla soluzione Google Home

**Reply by Jocman** [reply]
> Comunque non esiste solo Alexa... Dai un'occhiata anche a [QUESTO ARTICOLO](https://indomus.it/guide/local-fulfillment-integrare-localmente-google-assistant-a-home-assistant-addendum/)

Si lo avevo visto (grazie comunque della segnalazione), rimane pur sempre analogo alla questione cloud (seppur parziale) come Alexa.
Io sono "allergico" come detto, però devo dire che pure la mia metà ci mette del suo ("e se poi ci spiano?" 😄 😄 )
Comunque me lo guardo meglio

**Reply by snakuzzo** [reply]
no comment sulla frase della tua metà... a sto punto stacca internet perché l'assistente vocale è l'ultimo dei vostri problemi

**Reply by NdR91** [reply]
100€ magari no, visto che anche un Pi 0 W può fungere da satellite. Però si, parliamo di un sistema complesso che non ha ancora raggiunto un livello di applicazione al pari di quello che c'è oggi in commercio.

MNagari RhassPy in se funziona egregiamente come Assistente, ma il fatto di dover mettere in ogni stanza un Pi 0 W è già un limite, anche estetico se vogliamo. Non si può nemmeno nasconderlo eccessivamente, altrimenti ne risente la sensibilità nell'ascolto...

**Reply by Jocman** [reply]
> no comment sulla frase della tua metà... a sto punto stacca internet perché l'assistente vocale è l'ultimo dei vostri problemi

No, infatti. Il problema principale è la gente

**Reply by Jocman** [reply]
> Però si, parliamo di un sistema complesso che non ha ancora raggiunto un livello di applicazione al pari di quello che c'è oggi in commercio

Già , la considerazione da fare è anche quella. Di sicuro è interessante come prodotto, direi anche innovativo. Alcune cose che sto  organizzando per implementarle in domotica sono autocostruite (basate su ESP32), ma parliamo di ammenniccoli che vengono nascosti in scatole di derivazione/quadri elettrici, quindi esteticamente non (o poco) fastidiosi. In questo caso effettivamente ci sarebbe da darsi un bel pò da fare anche dal punto di vista estetico, non potendo essere "imboscate" più di tanto.
Penso che per il momento rimarrò sul commerciale (che sia amazon o google vedremo), magari in futuro a tempo perso cazzeggerò un pò con RhassPy

**Reply by M0ntyC0r4** [reply]
Rhasspy l'ho provato in tutte le salse; sono riuscito a farlo parlare con home assistant (hassio), anche da remoto. Il test (pulsante in alto a destra) va a buon fine, ho scritto le varie regole nella web ui, ho scelto di farlo parlare tramite intents, visto che li avevo già scritti per fare le mie prove col componente conversation...ma quando vado a dargli il comando (per ora per iscritto, con la spunta su "handle it") e poi clicco sul pulsante recognize, non mi dà errori rhasspy, non mi dà errori ha ma semplicemente non esegue il comando.

la parte casse, funziona: ho provato con delle casse bluetooth e parla.

la parte microfono non saprei, perché non ho un micfrono attaccato al raspberry e non sono riuscito a capire come far funzionare il microfono integrato nella cassa bluetooth.

**Reply by M0ntyC0r4** [reply]
Ho provato anche Genie, per la cronaca...un assistente vocale "multilingua" basato su almond.
Alla fine è quello che funziona meno peggio, anche se pure lui non capisce benissimo l'italiano parlato....anzi diciamo pure che ha grossi problemi di comprensione anche se alla fine, non so come né perché, pur capendo fischi per fiaschi finisce con l'eseguire correttamente il comando. Quindi diciamo che va meglio nell'interpretazione...funziona anche dall'app per smartphone, col microfono del cellulare, cosa che Rhasspy non mi sembra faccia (fra l'altro sarebbe scomodo perché dovresti passare per forza per la webui; infatti te lo dicono che è un assistente vocale offline).

**Comment by M0ntyC0r4** [comment]
Io ho provato a farlo con il servizio google_say e google_tts per le notifiche dell'assistente vocale su un mediaplayer a scelta ed il componente conversation, per impartirgli i comandi vocali.

Il giochetto funziona per iscritto (anche se devi scrivere righe e righe di codice per creare dei comandi in italiano e le sue conseguenti reazioni/risposte, altrimenti funzionano solo i comandi base turn_on/turn_off e pochi altri); tu scrivi il comando e l'assistente risponde una delle frasi che hai specificato tu e lo esegue.

Oralmente, quando clicchi sull'icona del microfono e gli dai il comando vocale, lui capisce fischi per fiaschi, perché non credo quella funzione sia ottimizzata per la lingua italiana.

Non so se può essere uno spunto interessante o se sapete come aggirare questo inconveniente, perché risolto questo l'assistente vocale ce l'hai e funziona...devi solo scrivertelo riga per riga!!!
